src/test/java/com/artipie/api/RepositoryVerticleTest.java

```diff
@@ -101,4 +124,35 @@ void listsUserRepos(final Vertx vertx, final VertxTestContext ctx) {
                 )
         );
     }
+
+    /**
+     * Waits until server port available.
+     *
+     * @param vertx Vertx instance
+     */
+    private static void waitServer(final Vertx vertx) {
+        final AtomicReference<Boolean> available = new AtomicReference<>(false);
+        final NetClient client = vertx.createNetClient();
+        final long max = System.currentTimeMillis() + RepositoryVerticleTest.MAX_WAIT.toMillis();
+        while (!available.get() && System.currentTimeMillis() < max) {
+            client.connect(
+                RepositoryVerticleTest.port, RepositoryVerticleTest.HOST,
+                ar -> {
+                    if (ar.succeeded()) {
+                        available.set(true);
+                    }
+                }
+            );
+            if (!available.get()) {
+                try {
+                    TimeUnit.MILLISECONDS.sleep(RepositoryVerticleTest.SLEEP_DURATION.toMillis());
+                } catch (final InterruptedException err) {
+                    break;
+                }
+            }
+        }
+        if (!available.get()) {
+            Assertions.fail();
```

Comment:
Maybe pass to `fail` a message which would explain what is wrong?

Reply:
@dgarus Thanks for good proposal.
I have added more informative message